tls/s2n_prf.c

```diff
@@ -205,11 +213,70 @@ static int s2n_evp_hmac_p_hash_free(struct s2n_prf_working_space *ws)
     return 0;
 }
 
+static const struct s2n_p_hash_hmac s2n_evp_pkey_hmac = {
+    .alloc = &s2n_evp_pkey_hmac_p_hash_new,
+    .init = &s2n_evp_pkey_hmac_p_hash_init,
+    .update = &s2n_evp_pkey_hmac_p_hash_update,
+    .final = &s2n_evp_pkey_hmac_p_hash_digest,
+    .reset = &s2n_evp_pkey_hmac_p_hash_reset,
+    .cleanup = &s2n_evp_pkey_hmac_p_hash_cleanup,
+    .free = &s2n_evp_pkey_hmac_p_hash_free,
+};
+#else
+static int s2n_evp_hmac_p_hash_alloc(struct s2n_prf_working_space *ws)
+{
+    POSIX_ENSURE_REF(ws->p_hash.evp_hmac.hmac_key = HMAC_CTX_new());
+    return 0;
+}
```

Comment:
I know you're just copying the old methods, but please use S2N_SUCCESS instead of 0.